src/ir_Carrier.cpp

```diff
@@ -182,6 +192,100 @@ bool IRrecv::decodeCarrierAC40(decode_results *results, uint16_t offset,
 }
 #endif  // DECODE_CARRIER_AC40
 
+#if SEND_CARRIER_AC48
+/// Send a CARRIER_AC48 formatted message.
+/// Alternative >64bit function to send CARRIER_AC48 messages
+/// Status: ALPHA / Untested.
+/// Where data is:
+///   uint8_t data[kCARRIER_AC48StateLength] = {0xA1, 0x20, 0x6B, 0xFF, 0xFF, 0xC5, 0x5E, 0xDF, 0x94, 0x00, 0x00, 0x3A};
+///
+/// @param[in] data An array of bytes containing the IR command.
+///                 It is assumed to be in MSB order for this code.
+/// @param[in] nbytes Nr. of bytes of data in the array. (>=kCARRIER_AC48StateLength)
+/// @param[in] repeat Nr. of times the message is to be repeated.
+void IRsend::sendCarrier_AC48(const uint8_t data[], const uint16_t nbytes, const uint16_t repeat) {
+  for (uint16_t r = 0; r <= repeat; r++) {
+    uint16_t pos = 0;
+    // Data Section #1
+    // e.g.
+    //   bits = 48; bytes = 6;
+    //   *(data + pos) = {0xA1, 0x20, 0x6B, 0xFF, 0xFF, 0xC5};
+    sendGeneric(kCarrier_AC48HdrMark, kCarrier_AC48HdrSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48OneSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48ZeroSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48SpaceGap,
+                data + pos, 6,  // Bytes
+                kCarrier_AC48Freq, true, kNoRepeat, kDutyDefault);
+    pos += 6;  // Adjust by how many bytes of data we sent
+    // Data Section #2
+    // e.g.
+    //   bits = 48; bytes = 6;
+    //   *(data + pos) = {0x5E, 0xDF, 0x94, 0x00, 0x00, 0x3A};
+    sendGeneric(kCarrier_AC48HdrMark, kCarrier_AC48HdrSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48OneSpace,
+                kCarrier_AC48BitMark, kCarrier_AC48ZeroSpace,
+                kCarrier_AC48BitMark, kDefaultMessageGap,
+                data + pos, 6,  // Bytes
+                kCarrier_AC48Freq, true, kNoRepeat, kDutyDefault);
+    pos += 6;  // Adjust by how many bytes of data we sent
```

Comment:
This appears to be two sections/messages. First normal, Second bit-wise inverted. We shouldn't store both in the "code"/message. The inverted copy/section should be calculated and sent on-the-fly.
e.g.
https://github.com/crankyoldgit/IRremoteESP8266/blob/494c5957626d3c92bbfd87bb73d3f04569cbd023/src/ir_Coolix.cpp#L81-L86

Reply:
@crankyoldgit do you know any good examples of protos where "extra data" is generated on the fly by send function (and likewise dropped by the decode) ?

Reply:
As we are experiencing, Midea is one such protocol.
e.g.
https://github.com/crankyoldgit/IRremoteESP8266/blob/8813010fbac0b33a9f2850f026a7b3303131cecf/src/ir_Midea.cpp#L67-L89

Goodweather:
https://github.com/crankyoldgit/IRremoteESP8266/blob/8813010fbac0b33a9f2850f026a7b3303131cecf/src/ir_Goodweather.cpp#L53-L59

Sharp:
https://github.com/crankyoldgit/IRremoteESP8266/blob/8813010fbac0b33a9f2850f026a7b3303131cecf/src/ir_Sharp.cpp#L79-L91